megamek/src/megamek/common/actions/PhysicalAttackAction.java

```diff
@@ -211,8 +211,23 @@ protected static void setCommonModifiers(ToHitData toHit, IGame game,
         }
 
         // if we're spotting for indirect fire, add +1
-        if (ae.isSpotting() && !ae.getCrew().hasActiveCommandConsole()) {
-            toHit.addModifier(+1, "attacker is spotting for indirect LRM fire");
+        if (ae.isSpotting() && !ae.getCrew().hasActiveCommandConsole()) {           
+            //but if it spots by TAG it doesn't.
+            boolean isSpotByTag = false;//temporal boolean flag
+            
+            //search through tagInfoForTurn and see if the entity has successfuly designated TAG on something.
+            Vector<TagInfo> tmpTagInfoForTurn = game.getTagInfo();
```

Comment:
This can be simplified a lot by using the foreach type of for loop:
```
for (TagInfo tmpInfo : game.getTagInfo()) {
    if (tmpInfo.attackerId == attackerId) {
        isSpotByTag = true;
        break;
    }
}